### PR TITLE
BUILD: macos - use matrix for builds, export proper cpu/arch when doi…

### PR DESCRIPTION
**Pull request by ciscon**
…ng builds

fixup

### CODE REVIEW

File: .github/workflows/scripts/homebrew.sh [hunk 1]
```diff
@@ -50,13 +51,18 @@ function install_arm64() {
 }
 
 function install_intel() {
+    export CPU="x86_64"
```
Comment (dsvensson) [inline .github/workflows/scripts/homebrew.sh:54]: CPU? Where is that env used?

Comment (dsvensson) [reply .github/workflows/scripts/homebrew.sh:54]: Aha, in the Makefile

Comment (ciscon) [reply .github/workflows/scripts/homebrew.sh:54]: righto, everything builds cleaner now, no need for the "just find a binary and name it 'this'" stuff anymore but meh.  ostensibly we could also make this a universal package by updating the run script, which we can think about.

Comment (dsvensson) [reply .github/workflows/scripts/homebrew.sh:54]: Ah, but then it shouldn't be in the install_* functions, as those only call brew, and brew only install precompiled packages.

File: .github/workflows/scripts/homebrew.sh [hunk 2]
```diff
@@ -66,10 +72,12 @@ function create_bundle() {
 }
 
 function build_intel() {
+    export CPU="x86_64"
     make strip
 }
 
 function build_arm64() {
+    export CPU="arm64"
```
Comment (dsvensson) [inline .github/workflows/scripts/homebrew.sh:80]: Shouldn't this be a make argument instead like DARWIN_TARGET?

**Pull request merged**